tests/conftest.py

```diff
@@ -72,6 +73,16 @@ def ring_idf():
 
     return IDF(StringIO(ring_idf_txt))
 
+@pytest.fixture()
```

Comment:
Please build the test case in code, rather than reading an IDF.

Reply:
I simplified the test code to build the case in code.